src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import java.util.List;
+
+/**
+ * Deals with interactions with the user.
+ */
+public class Ui {
+    private final TaskList taskList;
+
+    /**
+     * Constructor for Ui.
+     *
+     * @param taskList The task list to be displayed.
+     */
+    public Ui(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Shows upon exiting the program.
+     */
+    public String bye() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Displays the task list.
+     *
+     * @return task display.
+     */
+    public String displayTasks() {
+        String tasks = "Here are the tasks in your list:\n";
+
+        for (int i = 0; i < this.taskList.size(); i++) {
+            tasks += (i + 1) + ". " + taskList.get(i).toString() + "\n";
+        }
+        return tasks;
+    }
+
+    /**
+     * Notifies about the newly added task.
+     *
+     * @param task The new task added.
+     * @return Added task.
+     */
+    public String showAddedTask(Task task) {
+        assert (task != null);
+        String addedNotification = "Added: " + task.toString();
+        String taskSize = Integer.toString(this.taskList.size());
+        boolean isLessThanTwoTasks = this.taskList.size() <= 1;
+        String fullNotification = addedNotification + "\n"
+                + "Now you have " + taskSize + " task"
+                + (isLessThanTwoTasks ? "" : "s") + " in the list";
+        return fullNotification;
+    }
+
+    /**
+     * Notifies about the newly deleted task.
+     *
+     * @param toBeDeleted The task to be deleted.
+     * @return Deleted task.
+     */
+    public String showDeletedTask(Task toBeDeleted) {
+        assert (toBeDeleted != null);
```

Comment:
I like with how you use assert to simplify the code, but is it safe to use it for this public method ?